### PR TITLE
localnet scripts: use remote Docker images by default.

### LINKED ISSUE #328: Local testnet: scripts require building twice
The current instructions/scripts:

1. Produce a Docker image which requires a series of builds —and—
2. Expect to find a local `fendermint` binary for genesis targets.

This is wasteful and very expensive (requires downloading and building Cargo dependencies twice).

Ideally we'd use the `fendermint` binary inside the Docker image.

## Logs

```
$ cargo make --makefile ./infra/Makefile.toml testnode
[...]
 => exporting to image                                                                                                                0.5s
 => => exporting layers                                                                                                               0.4s
 => => writing image sha256:44a8eecf91868b2eb9075476226fd4c429b3f680252fdd2ae7e85101e5a19670                                          0.0s
 => => naming to docker.io/library/fendermint:latest                                                                                  0.0s
rm -rf docker/.artifacts
[cargo-make] INFO - Running Task: testnode-script-new-genesis
/tmp/fsio_DUxkOYXXDL.sh: 3: ./fendermint: not found
[cargo-make] ERROR - Error while executing command, exit code: 127
[cargo-make] WARN - Build Failed.
```

### PR DESCRIPTION
**Pull request by raulk**
## Context

Currently, these scripts build a Docker image. Simultaneously, they also expect a local Fendermint binary. It is strictly used to craft the genesis block. This is suboptimal because it requires duplicate compilation work, thus slowing down the user who just wants to get started quickly. The cargo graphs involved here are also quite heavy (1200+ packages), and some dependencies like rocksdb can take over 20min to compile.

## Changes

- Pull the Fendermint docker image from remote. We pull `fendermint:latest` by default, but can override the tag or switch to a local Docker image using env vars (see localnet.md doc). This makes it super fast to get started with IPC.
- Use Fendermint docker image for genesis and configuration, instead of a local fendermint binary. This solves the double build problem explained above.
- Removes code duplication between `testnet` and `testnode` targets.
- Revamp the localnet.md doc.

---

Fixes #328.

### CODE REVIEW

**Review by dnkolegov — APPROVED** [review]

**Comment by adlrocha** [comment]
Unless @dnkolegov, or @aakoshh want to do a final review here, I think we can merge this one (it'd unblock a few other PRs/lines of work). Thanks :pray:

**Pull request merged**
